tests/integration/test_apigateway.py

```diff
@@ -142,6 +142,15 @@ class TestAPIGateway:
     }
     TEST_API_GATEWAY_AUTHORIZER_OPS = [{"op": "replace", "path": "/name", "value": "test1"}]
 
+    @pytest.mark.aws_validated
+    def test_delete_rest_api_with_invalid_id(self, apigateway_client):
+        with pytest.raises(ClientError) as e:
```

Comment:
out of curiosity, the trap here isn't `NotFoundException` for a particular reason, is guaranteed not to be a `BadRequestException` ?